### DISCUSSION TITLE
use of Argument: Link to another node will not work

### DISCUSSION THREAD

**Discussion by Engesser** [freeplane/freeplane#513] [Ideas]
When I add an attribute to a node and tell the VALUE to jump to another node, it will show me the red arrow symbol, but only in the selection menu, not in the table.  I can jump , however, directly to a FILE to be opened, using this procedure (by listing the file name), but NOT to a NODE in the same mm-file.
This would be very valuable!

It would even be better if I could jump into another mm-file (eventually opening it) and hit the node of choice.

many greetings

Karl

**Comment by dpolivaev** [comment]
I am going to improve it in the coming version.

**Reply by Engesser** [reply]
Thank you Dimitry! Whats about my seceond question?: It would even be better if I could jump into another mm-file (eventually opening it) and hit the node of choice.

**Reply by dpolivaev** [reply]
I am sorry, I do not understand what you mean here.

**Reply by macmarrum** [reply]
@Engesser I believe you talk about #ID / #at notation for navigating to a node in another mindmap.

Please have a look at the exmaple map in #269

**Reply by dpolivaev** [reply]
Preview 1.10.4_07 supports the new reference format also in the attributes.

**Comment by macmarrum** [comment]
My tests show that a link to file, file#ID, file#at() show a red arrow, but a link to a node in the same map using #ID or #at() fail to show a (green) arrow.

[bug-links_in_attributes.zip](https://github.com/freeplane/freeplane/files/9332934/bug-links_in_attributes.zip)

**Reply by Engesser** [reply]
hello Dimitry, I don`t understand: Did you correct the behavour of freeplane? What I am going to do is: add a series of attributes to a node and use for each individual attribute a link to another node in the same mm.file. A link which I can click on, causing an automatic jump to the other node.  It would be even better if I could add a link to a node even in another mm.file!
This is what I would like to have.
As there is no update I assume nothing have changed up to now?

many greetings

Karl

**Reply by Engesser** [reply]
ADDENDUM: now I got you, sorry! I downloaded https://jztkft.dl.sourceforge.net/project/freeplane/freeplane%20preview/1.10.x/Freeplane-Setup-with-Java-1.10.4-pre07.exe and installed it. First I used a version without JAVA (I thought I already istalled JAVE?!) but was constantly asked to install JAVA/OK, after starting the programme I got a shock: GIANTIC Menueformat! Is there a way to reduce the size of the menu items to normal values?

Greetings

Karl

**Comment by Engesser** [comment]
![RIESENANSICHT_FREEPLANE_nach_Neuinstallation_Preview](https://user-images.githubusercontent.com/109984984/184658549-882f37ed-f1ac-4254-b720-9196b6e1681c.PNG)

**Reply by dpolivaev** [reply]
Create a new issue (not a discussion topic) and upload your auto.properties so that I can check it.

**Reply by Engesser** [reply]
hello Dimitry, I just added all the possibly relevant files of my installation into a Download Container, hopefully this will help? The problem is that I cannot work at the moment. Should I step back to an older version?

https://www.dropbox.com/sh/co0camvs7q24iip/AAA9D_V9K0oURN9rku30alF0a?dl=0

Greetings

Karl

**Reply by macmarrum** [reply]
I can see in the file `auto.properties` the following line
```properties
menuItemFontSize=100
```
This is the reason for huge fonts.

This option can be set to a smaller value in `Preferences…->Appearance->Look and feel->Menu item font size`, but it might be a difficult task in the current state. 
Here's a workaround:
1. Close Freeplane
2. Edit `C:\Users\Engadmin\AppData\Roaming\Freeplane\1.10.x\auto.properties` in e.g. notepad.exe and change `menuItemFontSize=100` to `menuItemFontSize=10`
3. Save the file and close the editor (notepad.exe)
4. Start Freeplane

**Reply by Engesser** [reply]
**That helped immediately! Thanks a lot, Dimitry!!**

**Reply by dpolivaev** [reply]
All thanks go to @macmarrum

**Reply by Engesser** [reply]
Thanks to macmarrum also! I overlooked this, sorry!
Greetings

Karl

**Reply by Engesser** [reply]
one small problem remains:

the update did not only change the menue font size but also other styles, e.g. FORMAT of nodes etc./May be in the Font-Chaos I did a mistake?
Now my question: 
Can I theoretically use an older version ( I hope I have a backup somewhere?) of the auto.properties file and copy it into? Or should I simply reinstall the preview version?
many greetings, Karl

**Reply by dpolivaev** [reply]
There was indeed a bug related to the `menuItemFontSize=100` which I am fixing in the coming preview, but after you removed this line from the properties you should be fine.
I do not understand what exact changes happened in the node formats. You can upload your screenshots to show it.
I do not see how the older version could help.
Generally, if you try preview versions is good if you have some backups of your maps just in case something goes wrong.

**Comment by Engesser** [comment]
Dear Dimitry, dear Macmarrum,

I don't want to push, but is there an answer?

greetings,   Karl

By the way, the improved version does now show the desired  links in the attributes! Thank you all!

**Reply by macmarrum** [reply]
> Can I theoretically use an older version ( I hope I have a backup somewhere?) of the auto.properties file and copy it into?

My guess is that Freeplane should be fine using `auto.properties` from an older version. There might have been some format changes for some properties, and they would be lost.
The only way to find out is to try it, unfortunately. So frequent backup is your friend :wink:

**Comment by Engesser** [comment]
hello Macmarrum,

yes, I will test. The strange thing is, that I cannot change nothing, e.g. the node style. It is simply set to STANDARD. And I cannot change it, its greyed out, also in the menu item FORMAT.
May be I set accidentally STANDARD-1.6.mm as a standard for all nodes?
Greetings
Karl

**Reply by dpolivaev** [reply]
If you can not edit the maps it looks like you have activated the viewing mode. 
To change it back use Edit->Editing

**Comment by Engesser** [comment]
Dear Dimitry, you just saved me from fresh installation of the programme! Naturally that was the solution, I did not know that Freeplane has a "All-locked"- mode? Thanks!

**Comment by Engesser** [comment]
Some CLARIFICATION possible?

Normal procedure: Having an argument and in the value I add a node by double clicking on it, then RETURN.

When I use the argument and would like to use as a VALUE not only the #ID number, but some TEXT after the ID, this seems to be impossible? Freeplane will show the green arrow after double click only when I remove all text behind the ID number?! Is this indeed the case? Or do I make a mistake?

Greetings

Heleus/Karl

**Reply by dpolivaev** [reply]
Yes, and you can use two different attributes for your text and for your link

**Reply by Engesser** [reply]
hello Dimitry,
may be I didn`t understand this? 
if I am right, then for each Attribut-line there exists ONE VALUE? OK, you mean if I will set a link as a "value" then this line will contain ONLY the link and the explanative Text I have to add in a separate ATTRIBUTE line?!

Greetings

Karl

**Reply by Engesser** [reply]
hello, still waiting for a reply? It is not possible to add an explanative text AFTER the ID_Number?! I would just like to be sure!
Greetings
Karl

**Reply by dpolivaev** [reply]
No, it is not possible now.
If you want to increase the chance that it becomes implemented, please explain why it is important and share your examples.

**Reply by macmarrum** [reply]
Hi @Engesser

As far as I understand you'd expect a way to add an attribute containing a link with custom text, like it can be done in Details or Note

![2022-09-11_14,45,47](https://user-images.githubusercontent.com/7197131/189528275-0baa8a10-3a0c-497b-b6de-ac4e96615a01.png)

When I set a link to a node in an attribute (`right-click + Set Node Link...`), I can see the target node's text displayed.
When I double-click the attribute value, I can see the node id (#ID_876946455), and on the drop-down list I can choose a link (red-arrow icon) or text (ABC icon).

![2022-09-11_14,40,42](https://user-images.githubusercontent.com/7197131/189528120-fdbc3f60-8735-458e-9a1c-20272f2488d1.png)

The nice thing about attributes is that they have a key and a value, so I can put my descriptive text in the key, like "link to root"

![2022-09-11_14,53,27](https://user-images.githubusercontent.com/7197131/189528556-90abe54a-dc30-4653-9443-ab26cbb448e1.png)

If this is not enough for you needs, please consider using HTML links in Details or Note.

**Reply by euu2021** [reply]
It's also possible to use an alias. So, 3 ways to link bbbbb from aaaaa properties (ccc being the alias for bbbbb):

![image](https://user-images.githubusercontent.com/77707706/189555416-8ffdcdb6-45eb-4240-bf48-769f8a29afe5.png)

But, the alias would only be useful if Engesser wants the same text every time he will link to the node, which, probably, is not the case, right?

**Comment by Engesser** [comment]
Hi Macmarrum,

Yes, it is possible to create a link directly with the synthax "#at(:~myNodeName)". But I think that you can create only ONE link this way, converting the red link (file reference) to a GREEN arrow doesn't seem to work this way. The ALIAS will not help in this case?
With the ATTRIBUTES, on the other hand, it works perfectly. Your tip was impressively simple, I just did not get it: I can write the node name also in the ATTRIBUTNAMEN! Nevertheless it would be good if one could write also in the VALUE area still behind the ID number descriptive text? Let's see if Dimitry finds time for that?

many greetings

Karl

**Reply by euu2021** [reply]
What do you want to write in the VALUE field? I see that you want "DESCRIPTIVE TEXT" + "ID NUMBER", but what exactly is this descriptive text? Is it always the same when the node is linked, or it changes for each instance? If it's always the same, you can use the alias. For example, I will link node B into node the attributes of node A, and I have "DESCRIPTIVE TEXT" + "ID NUMBER", with a green arrow:

![image](https://user-images.githubusercontent.com/77707706/190280674-968327fc-6699-4c0d-b811-7b71927d92ce.png)

**Comment by dpolivaev** [comment]
Hello Karl, if you want to increase the chance that your request is considered and implemented you can do more than describe what you want. You can describe why you want it. The more understanding of your use case I can get and the better reasons you find, the more likely the feature is implemented.

Because if I implement features without an excellent understanding of concrete needs, I am not likely to find good solutions useful for many people.

**Reply by Engesser** [reply]
hello euu2021, it changes. I add a screenshot how it looks like?
![Freeplane_multiple_LINKS_in_attributes](https://user-images.githubusercontent.com/109984984/190385608-2bc0b1d9-2b27-4e7e-9111-2f4dc0dabe92.PNG)

**Reply by euu2021** [reply]
I see that each of your nodes have a different description text when they are linked. But, the question is: each time a node is linked the description is the same? (in your screenshot, the answer is yes)

Another question: is the description text equal to the node core text? (in your screenshot, the answer is yes)

**Reply by dpolivaev** [reply]
The attached screenshot does not explain the use case and it does not explain the reason for the request.
I need to see a REAL EXAMPLE and no testing garbage.

Regards.
